lib/ex_unit/lib/ex_unit/pattern/diff_context.ex

```diff
@@ -0,0 +1,43 @@
+defmodule ExUnit.Pattern.DiffContext do
+  alias ExUnit.{ContainerDiff, PatternDiff}
+
+  defstruct [:keys, :print_when?, :vars, :pins]
```

Comment:
The attribute `print_when?` can also be dropped if you have two separated diffs because  you will only have `when` when they actually exists and should be printed